src/java/main/org/apache/zookeeper/common/ZKConfig.java

```diff
@@ -0,0 +1,221 @@
+        addConfiguration(configFile);
+    }
+
+    private void init() {
+        /**
+         * backward compatibility for all currently available client properties
+         */
+        handleBackwardCompatibility();
+    }
+
+    /**
+     * Now onwards client code will use properties from this class but older
+     * clients still be setting properties through system properties. So to make
+     * this change backward compatible we should set old system properties in
+     * this configuration.
+     */
+    protected void handleBackwardCompatibility() {
+        properties.put(SSL_KEYSTORE_LOCATION, System.getProperty(SSL_KEYSTORE_LOCATION));
+        properties.put(SSL_KEYSTORE_PASSWD, System.getProperty(SSL_KEYSTORE_PASSWD));
+        properties.put(SSL_TRUSTSTORE_LOCATION, System.getProperty(SSL_TRUSTSTORE_LOCATION));
+        properties.put(SSL_TRUSTSTORE_PASSWD, System.getProperty(SSL_TRUSTSTORE_PASSWD));
+        properties.put(SSL_AUTHPROVIDER, System.getProperty(SSL_AUTHPROVIDER));
+        properties.put(JUTE_MAXBUFFER, System.getProperty(JUTE_MAXBUFFER));
+        properties.put(KINIT_COMMAND, System.getProperty(KINIT_COMMAND));
+        properties.put(JGSS_NATIVE, System.getProperty(JGSS_NATIVE));
+    }
+
+    /**
+     * Get the property value
+     *
+     * @param name
+     * @return property value
+     */
+    public String getProperty(String name) {
+        return properties.get(name);
+    }
+
+    /**
+     * Get the property value, if it is null return default value
+     *
+     * @param name
+     *            Property name
+     * @param defaultValue
+     * @return
+     */
+    public String getProperty(String name, String defaultValue) {
+        String value = properties.get(name);
+        if (value == null) {
+            return defaultValue;
+        } else {
+            return value;
+        }
+    }
+
+    /**
+     * Return the value of "java.security.auth.login.config" system property
+     *
+     * @return value
+     */
+    public String getJaasConfKey() {
+        return System.getProperty(Environment.JAAS_CONF_KEY);
+    }
+
+    /**
+     * Set the property value
+     *
+     * @param name
+     * @param value
+     */
+    public void setProperty(String key, String value) {
+        if (null == key) {
+            throw new IllegalArgumentException("property key is null.");
+        }
+        String oldValue = properties.put(key, value);
+        if (LOG.isDebugEnabled()) {
+            if (null != oldValue && !oldValue.equals(value)) {
+                LOG.debug("key {}'s value {} is replaced with new value {}", key, oldValue, value);
+            }
+        }
+    }
+
+    /**
+     * Add a configuration resource. The properties form this configuration will
+     * overwrite corresponding already loaded property and system property
+     *
+     * @param configFile
+     *            Configuration file.
+     */
+    public void addConfiguration(File configFile) throws ConfigException {
```

Comment:
It might not be a bad idea to add it, although we don't seem to need it right now, so we could also leave it for later if needed. What do you think?